flow/flow.loc.asgn/tests/testthat/test-wrap-loc-asgn.R

```diff
@@ -0,0 +1,200 @@
+
+#' @keywords Currently none
+
+#' @examples
+#' Not run
+#' wrap.loc.asgn(DirIn="pfs/location_asset/ptb330a/10312",
+#'              DirOutBase="pfs/out",
+#'              TimeBgn=as.POSIXct('2019-01-01',tz='GMT),
+#'              TimeEnd=as.POSIXct('2019-06-01',tz='GMT),
+#'              TypeFile='asset'
+#'              )
+
+#' @seealso None
+#'
+# changelog and author contributions / copyrights
+#   Mija Choi (2021-04-19)
+#     Original Creation
+##############################################################################################
+# Define test context
+context("\n                       Unit test of wrap.loc.asgn.R\n")
+
+# Unit test of wrap.loc.asgn.R
+test_that("Unit test of wrap.loc.asgn.R", {
+  source('../../wrap.loc.asgn.R')
+  library(stringr)
+  
+  wk_dir <- getwd()
+  testOutputDir = "pfs/out"
+  
+  # Test scenario 1:: within the valid time range
+  # 10312 does not have "active_periods" pass TypeFile = 'asset'
+  testInputDir <- base::paste0(wk_dir, '/', 'pfs/location_asset/ptb330a/10312')
+  fileLoc <- base::dir(testInputDir)
+  
+  # Load in the location json and get the location name to verify the test
+  loc <- NEONprocIS.base::def.loc.meta(NameFile = base::paste0(testInputDir,'/',fileLoc))
+  installdate <- str_replace_all(loc$install_date, "-", "/")
+  testOutputDirPath <- base::paste0(testOutputDir,"/",loc$source_type,"/",installdate,"/",loc$source_id,"/location",collapse='/')
+  
+  # clean out the test output dirs and file recursively
+  if (dir.exists(testOutputDir)) {
+    unlink(testOutputDir, recursive = TRUE)
+  }
+  returnedOutputDir <- wrap.loc.asgn(
+    DirIn = testInputDir,
+    DirOutBase = testOutputDir,
+    TimeBgn = as.POSIXct('2019-01-01', tz = 'GMT'),
+    TimeEnd = as.POSIXct('2019-01-06', tz = 'GMT'),
+    TypeFile = 'asset'
+  )
+  
+  testthat::expect_true (dir.exists(testOutputDirPath))
+  #
+  # Test scenario 2:: within the valid time range
+  # 100959 has "active_periods" pass TypeFile = 'namedLocation'
+  
+  testInputDir <- base::paste0(wk_dir, '/', 'pfs/location_asset/ptb330a/100959')
+  fileLoc <- base::dir(testInputDir)
+  
+  # Load in the location json and get the location name to verify the test
+  loc <- NEONprocIS.base::def.loc.meta(NameFile = base::paste0(testInputDir, '/', fileLoc))
+  installdate <- str_replace_all(as.Date(loc$install_date), "-", "/")
+  testOutputDirPath <- base::paste0(testOutputDir,"/",loc$source_type,"/",installdate,"/",loc$source_id,"/location",collapse = '/')
+  
+  # clean out the test output dirs and file recursively
+  if (dir.exists(testOutputDir)) {
+    unlink(testOutputDir, recursive = TRUE)
+  }
+  returnedOutputDir <- wrap.loc.asgn(
+    DirIn = testInputDir,
+    DirOutBase = testOutputDir,
+    TimeBgn = as.POSIXct('2019-01-01', tz = 'GMT'),
+    TimeEnd = as.POSIXct('2019-01-06', tz = 'GMT'),
+    TypeFile = 'namedLocation'
+  )
+  testthat::expect_true (dir.exists(testOutputDirPath))
```

Comment:
This test fails because the location file does not exist. But I think this test is a duplicate of Test Scenario 1. Delete?

Reply:
1. Oh, I forgot to add the file, /locations/prt/24688/prt_24688_locations.json.  But doesn't look like it is needed for the test. 

2. For the test 3, I modified it to be a negative; in case TypeFile = 'namedLocation' but the input json has missing "active_dates" Let me know if this is not necessary.  (this error is not caught atm.)

3.  checking the presence of the output file is added